cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -767,42 +810,6 @@ class FlowSummaryNode extends Node, TFlowSummaryNode {
   override string toStringImpl() { result = this.getSummaryNode().toString() }
 }
 
-/**
- * INTERNAL: do not use.
- *
- * A node representing an indirection of a parameter.
- */
-class IndirectParameterNode extends Node instanceof IndirectInstruction {
-  InitializeParameterInstruction init;
-
-  IndirectParameterNode() { IndirectInstruction.super.hasInstructionAndIndirectionIndex(init, _) }
-
-  int getArgumentIndex() { init.hasIndex(result) }
-
-  /** Gets the parameter whose indirection is initialized. */
-  Parameter getParameter() { result = init.getParameter() }
-
-  override Declaration getEnclosingCallable() { result = this.getFunction() }
-
-  override Declaration getFunction() { result = init.getEnclosingFunction() }
-
-  /** Gets the underlying operand and the underlying indirection index. */
-  predicate hasInstructionAndIndirectionIndex(Instruction instr, int index) {
-    IndirectInstruction.super.hasInstructionAndIndirectionIndex(instr, index)
-  }
-
-  override Location getLocationImpl() { result = this.getParameter().getLocation() }
```

Comment:
This no longer occurs after the changes. Is that on purpose?

Reply:
Oh, well spotted. Yeah, there wasn't really a reason to override this as the `getLocationImpl` predicate has a perfectly reasonable implementation from `IndirectInstruction`. In fact, when I replaced this call with `IndirectInstruction.super.getLocationImpl()` ql-for-ql told me that it was redundant so I simply deleted it.

As a sanity check I ran:
```ql
private predicate foo(ParameterNode n) {
  not exists(n.getLocation())
}
```
on a DB I had locally, and got no results back 🎉